Engine/Login/LoginCliAction.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using OpenTap.Cli;
+
+namespace OpenTap.Login
+{
+    /// <summary>
+    /// Logs in to an Keycloak instance.
+    /// </summary>
+    [Display("login", Group: "web")]
+    public class LoginCliAction : ICliAction
+    {
+        /// <summary> The user name</summary>
+        [CommandLineArgument("username")] public string UserName { get; set; }
+        /// <summary> The password</summary>
+        [CommandLineArgument("password")] public string Password { get; set; }
+        /// <summary> The URL for the keycloak instance</summary>
+        [CommandLineArgument("url")] public string Url { get; set; }
+        /// <summary> the client ID </summary>
+        [CommandLineArgument("client-id")] public string ClientId { get; set; }
+        /// <summary> The realm. </summary>
+        [CommandLineArgument("realm")] public string Realm { get; set; }
+
+        /// <summary> This can be used if the url does not contain the service site. </summary>
+        [CommandLineArgument("site")] public string Site { get; set; }
+
+        private static readonly TraceSource log = Log.CreateSource("web");
+
+        /// <summary> Executes the action. </summary>
+        public int Execute(CancellationToken cancellationToken)
+        {
+            var url = $"https://{Url}/auth/realms/{Realm}/protocol/openid-connect/token";
```

Comment:
`auth/realms` seems specific for Keycloak.
`https://{Url}/auth/realms/{Realm}/` seems generic however.